sys/net/application_layer/uhcp/uhcp.c

```diff
@@ -97,14 +97,17 @@ void uhcp_handle_push(uhcp_push_t *req, uint8_t *src, uint16_t port, uhcp_iface_
     char addr_str[INET6_ADDRSTRLEN];
     char prefix_str[INET6_ADDRSTRLEN];
     inet_ntop(AF_INET6, src, addr_str, INET6_ADDRSTRLEN);
-    uint8_t prefix[16];
+    uint8_t prefix[IN6ADDRSZ] = { 0 };
```

Comment:
this is an implicit call to memset, maybe move below the sanity check?

Reply:
When I apply the following patch

```diff
diff --git a/sys/net/application_layer/uhcp/uhcp.c b/sys/net/application_layer/uhcp/uhcp.c
index 216168e843..35d5f40440 100644
--- a/sys/net/application_layer/uhcp/uhcp.c
+++ b/sys/net/application_layer/uhcp/uhcp.c
@@ -97,10 +97,11 @@ void uhcp_handle_push(uhcp_push_t *req, uint8_t *src, uint16_t port, uhcp_iface_
     char addr_str[INET6_ADDRSTRLEN];
     char prefix_str[INET6_ADDRSTRLEN];
     inet_ntop(AF_INET6, src, addr_str, INET6_ADDRSTRLEN);
-    uint8_t prefix[IN6ADDRSZ] = { 0 };
+    uint8_t prefix[IN6ADDRSZ];
     if (req->prefix_len == 0) {
         return;
     }
+    prefix = { 0 };
     size_t prefix_bytes = (req->prefix_len + 7)>>3;
     memcpy(prefix, req->prefix, prefix_bytes);
 
```


I get an error:
```
/home/mlenders/Repositories/RIOT-OS/RIOT/sys/net/application_layer/uhcp/uhcp.c: In function 'uhcp_handle_push':
/home/mlenders/Repositories/RIOT-OS/RIOT/sys/net/application_layer/uhcp/uhcp.c:104:14: error: expected expression before '{' token
  104 |     prefix = { 0 };
      |    
```

So rather keep as is? I prefer this notation over an explicit `memset()`.

Reply:
Yeah, I was thinking of moving the whole "uint8_t prefix[IN6ADDRSZ] = { 0 };" to *after* the check. Anyhow, not really a live-changing improvement, so of course let's keep it and sorry for causing trouble.

Reply:
I also prefer to keep variable declarations at the top of the block they are used in ;-). I kept `prefix_bytes` only where it is, because it already was there.